apps/server/src/modules/course/api/course-info.uc.ts

```diff
@@ -78,43 +78,31 @@ export class CourseInfoUc {
 		return courseInfos;
 	}
 
-	private async getSyncedGroupName(groupId: EntityId): Promise<string> {
-		const group = await this.groupService.findById(groupId);
+	private async getSyncedGroupName(groupId: EntityId): Promise<string | undefined> {
+		const group = await this.groupService.tryFindById(groupId);
 
-		return group.name;
+		return group?.name;
 	}
 
 	private async getCourseTeacherFullNames(teacherIds: EntityId[]): Promise<string[]> {
-		const teacherNames = await Promise.all(
-			teacherIds.map(async (teacherId): Promise<string> => {
-				const teacher = await this.userService.findById(teacherId);
-				const fullName = teacher.firstName.concat(' ', teacher.lastName);
+		const teachers = await Promise.all(teacherIds.map((id) => this.userService.findByIdOrNull(id)));
 
-				return fullName;
-			})
-		);
-		return teacherNames;
+		return teachers
+			.filter((teacher): teacher is UserDo => Boolean(teacher))
+			.map((teacher) => teacher.firstName.concat(' ', teacher.lastName));
```

Comment:
```
const promises = teacherIds.map((id) => this.userService.findByIdOrNull(id));
const teachers = await Promise.all(promises);
const existingTeacherFullNames = teachers
			.filter((teacher): teacher is UserDo => Boolean(teacher))
			.map((teacher) => teacher.firstName.concat(' ', teacher.lastName))
			
return existingTeacherFullNames;
```

By all others it can be structured in the same way.